src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseCancellationTokenThrowIfCancellationRequested.Fixer.cs

```diff
@@ -0,0 +1,132 @@
+                //      if (token.IsCancellationRequested)
+                //          throw new OperationCanceledException();
+                //      else
+                //          Frob();
+                //  Replace with:
+                //      token.ThrowIfCancellationRequested();
+                //      Frob();
+                createChangedDocument = async token =>
+                {
+                    var editor = await DocumentEditor.CreateAsync(context.Document, token).ConfigureAwait(false);
+                    SyntaxNode expressionStatement = CreateThrowIfCancellationRequestedExpressionStatement(editor, conditional, propertyReference);
+                    editor.ReplaceNode(conditional.Syntax, expressionStatement);
+
+                    if (conditional.WhenFalse is IBlockOperation block)
+                    {
+                        editor.InsertAfter(expressionStatement, block.Operations.Select(x => x.Syntax.WithAdditionalAnnotations(Formatter.Annotation)));
+                    }
+                    else if (conditional.WhenFalse is not null)
+                    {
+                        editor.InsertAfter(expressionStatement, conditional.WhenFalse.Syntax);
+                    }
+
+                    return editor.GetChangedDocument();
+                };
+            }
+            else if (symbols.IsNegatedCheckWithThrowingElseClause(conditional, out propertyReference))
+            {
+                //  For negated checks of the form:
+                //      if (!token.IsCancellationRequested) { DoStatements(); }
+                //      else { throw new OperationCanceledException(); }
+                //  Replace with:
+                //      token.ThrowIfCancellationRequested();
+                //      DoStatements();
+                createChangedDocument = async token =>
+                {
+                    var editor = await DocumentEditor.CreateAsync(context.Document, token).ConfigureAwait(false);
+                    SyntaxNode expressionStatement = CreateThrowIfCancellationRequestedExpressionStatement(editor, conditional, propertyReference)
+                        .WithAdditionalAnnotations(Formatter.Annotation);
+                    editor.ReplaceNode(conditional.Syntax, expressionStatement);
+
+                    if (conditional.WhenTrue is IBlockOperation block)
+                    {
+                        editor.InsertAfter(expressionStatement, block.Operations.Select(x => x.Syntax.WithAdditionalAnnotations(Formatter.Annotation)));
+                    }
+                    else
+                    {
+                        editor.InsertAfter(expressionStatement, conditional.WhenTrue.Syntax);
+                    }
+
+                    //  Ensure if-blocks with multiple statements maintain correct indentation.
+                    return await Formatter.FormatAsync(editor.GetChangedDocument(), Formatter.Annotation, cancellationToken: token).ConfigureAwait(false);
```

Comment:
@sharwell Is this the recommend way to achieve this formatting? Normally, I see fixers just attaching the formatter annotation and not explicitly invoking the formatter API. @NewellClark Any reason that approach did not work?

Reply:
@mavasani I did not realize that `editor.GetChangedDocument()` would automatically apply formatting. That's good to know. I've gone ahead and simplified it.